nix/keys.nix

```diff
@@ -6,13 +6,34 @@ let
   keyOptionsType = types.submodule {
     options.text = mkOption {
       example = "super secret stuff";
-      type = types.str;
+      default = null;
+      type = types.nullOr types.str;
       description = ''
-        The text the key should contain. So if the key name is
-        <replaceable>password</replaceable> and <literal>foobar</literal>
-        is set here, the contents of the file
+        When non-null, this designates the text that the key should
+        contain. So if the key name is <replaceable>password</replaceable>
+        and <literal>foobar</literal> is set here, the contents of the file
         <filename>/run/keys/<replaceable>password</replaceable></filename>
         will be <literal>foobar</literal>.
+
+        NOTE: Either <literal>text</literal> or <literal>filename</literal> have
+        to be set.
+      '';
+    };
+
+    options.filename = mkOption {
+      example = "./super-secret.key";
+      default = null;
+      type = types.nullOr types.str;
```

Comment:
No, it's with XML queries for non-`str` types being harder, and unprecedented (according to my, admittedly limited, grep skill) in the nixops repository.  I could get down to the bottom of matter, but it would take more effort.